cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/TransformContext.java

```diff
@@ -88,4 +88,9 @@
    * @throws UnsupportedOperationException if the program is not running under the adapter context
    */
   <T> Class<T> loadPluginClass(String pluginId);
+
+  /**
+   * @return the {@link LookupProvider} implementation, for getting lookup tables
+   */
+  LookupProvider getLookupProvider();
```

Comment:
You can keep most of the implementation, only need to have `AbstractTransformContext` takes a `LookupProvider` instance in the constructor and delegates to that when implementing methods in `LookupProvider`.